compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1384,23 +1384,32 @@ public void visit(BLangErrorType errorTypeNode) {
             errorTypeNode.flagSet.add(Flag.ANONYMOUS);
         }
 
+        // The builtin error type
+        BErrorType bErrorType = symTable.errorType;
+
         boolean distinctErrorDef = errorTypeNode.flagSet.contains(Flag.DISTINCT);
         if (detailType == symTable.detailType && !distinctErrorDef &&
                 !this.env.enclPkg.packageID.equals(PackageID.ANNOTATIONS)) {
-            resultType = symTable.errorType;
+            resultType = bErrorType;
             return;
         }
 
         // Define user define error type.
-        BErrorTypeSymbol errorTypeSymbol = Symbols
-                .createErrorSymbol(Flags.asMask(errorTypeNode.flagSet), Names.EMPTY, env.enclPkg.symbol.pkgID,
-                                   null, env.scope.owner, errorTypeNode.pos, SOURCE);
+        BErrorTypeSymbol errorTypeSymbol;
+
+        errorTypeSymbol = Symbols
+                .createErrorSymbol(Flags.asMask(errorTypeNode.flagSet), Names.EMPTY, bErrorType.tsymbol.pkgID, null,
+                        bErrorType.tsymbol.owner, errorTypeNode.pos, SOURCE);
 
         PackageID packageID = env.enclPkg.packageID;
         if (env.node.getKind() != NodeKind.PACKAGE) {
             errorTypeSymbol.name = names.fromString(
                     anonymousModelHelper.getNextAnonymousTypeKey(packageID));
             symbolEnter.defineSymbol(errorTypeNode.pos, errorTypeSymbol, env);
+        } else {
+            // If the error type descriptor is at the module level, the error type symbol's owner should be the current
+            // environment's scope owner.
+            errorTypeSymbol.owner = env.scope.owner;
```

Comment:
This is problematic for module level defined error type vars. 
e.g.,
```ballerina
<cursor1>error<DetailType> err1;

function test() {
    <cursor2>error<DetailType> err2;
}
```
The module returned for the above 2 cases will be,
- the default(current) module for `cursor1`
- the `lang.annotations` module for `cursor2`.

For both cases above it should've returned the `lang.annotations` module.

Reply:
Handled the issue caused by this in the `SymbolEnter`.